### PR TITLE
wip: Add relaunch and recover to flytekit remote

### PR DESCRIPTION
**Pull request by katrogan**
Signed-off-by: Katrina Rogan <katroganGH@gmail.com>

# TL;DR
Add relaunch and recover to flytekit remote

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1420

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit/remote/remote.py [hunk 1]
```diff
@@ -671,6 +671,66 @@ def execute(
         """
         raise NotImplementedError(f"entity type {type(entity)} not recognized for execution")
 
+    def relaunch(
+        self,
+        execution: FlyteWorkflowExecution,
+        execution_name: str = None,
```
Comment (wild-endeavor) [inline flytekit/remote/remote.py:677]: typing.Optional

File: flytekit/remote/remote.py [hunk 2]
```diff
@@ -671,6 +671,66 @@ def execute(
...
+        relaunched_execution = FlyteWorkflowExecution.promote_from_model(self.client.get_execution(exec_id))
+
+        if wait:
+            return self.wait(relaunched_execution)
+        return relaunched_execution
+
+    def recover(
+        self,
+        execution: FlyteWorkflowExecution,
+        execution_name: str = None,
```
Comment (wild-endeavor) [inline flytekit/remote/remote.py:705]: typing.Optional

File: flytekit/remote/remote.py (hunk 1 above)
Comment (katrogan) [reply flytekit/remote/remote.py]: done

File: flytekit/remote/remote.py (hunk 2 above)
Comment (katrogan) [reply flytekit/remote/remote.py]: done

**Comment by EngHabu** [comment]
Are we merging this for 0.17.0, @katrogan ?

**Pull request closed without merging**